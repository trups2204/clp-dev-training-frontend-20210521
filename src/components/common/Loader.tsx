import React from 'react';
import {ScreenReaderOnly,SpinnerPanel, SpinnerSvg, SvgCircle} from './styles';

export const Loader =() =>{
    return (
        <SpinnerPanel data-test-id="loading" role="alert" aria-live="polite" aria-busy="true" aria-atomic="true">
            <SpinnerSvg >
                <SvgCircle cx="30" cy="30" r="20"  />
            </SpinnerSvg>
            <ScreenReaderOnly>Please wait..Page is Loading</ScreenReaderOnly>
        </SpinnerPanel>
    );
};