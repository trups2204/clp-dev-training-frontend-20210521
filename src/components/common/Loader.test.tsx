import React from 'react';
import renderer from 'react-test-renderer';
import  {Loader} from './Loader';

describe('Loader test', ()=>{

    it('loads loader component',()=>{

        const component = <Loader/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    })
})