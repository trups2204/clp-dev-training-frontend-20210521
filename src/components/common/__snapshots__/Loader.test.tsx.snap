// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader test loads loader component 1`] = `
<div
  aria-atomic="true"
  aria-busy="true"
  aria-live="polite"
  className="sc-eCApnc lnjSS"
  data-test-id="loading"
  role="alert"
>
  <svg
    className="sc-jSFjdj brjlgd"
  >
    <circle
      className="sc-gKAaRy jgrAZo"
      cx="30"
      cy="30"
      r="20"
    />
  </svg>
  <p
    className="sc-iCoGMd boJwNj"
  >
    Please wait..Page is Loading
  </p>
</div>
`;
